src/contracts/yas_pool.cairo

```diff
@@ -84,25 +185,304 @@ mod YASPool {
+                    self.fee.read()
+                );
+
+                state.sqrt_price_X96 = ret_sqrt_price_X96;
+
+                if exact_input {
+                    state
+                        .amount_specified_remaining -=
+                            IntegerTrait::<i256>::new(step_amount_in + step_fee_amount, false);
+                    state.amount_calculated = state.amount_calculated
+                        - IntegerTrait::<i256>::new(step_amount_out, false);
+                } else {
+                    state
+                        .amount_specified_remaining +=
+                            IntegerTrait::<i256>::new(step_amount_out, false);
+                    state.amount_calculated = state.amount_calculated
+                        + IntegerTrait::<i256>::new(step_amount_in + step_fee_amount, false);
+                };
+
+                // if the protocol fee is on, calculate how much is owed, decrement feeAmount, and increment protocolFee
+                if cache.fee_protocol > 0 {
+                    let delta: u256 = step_fee_amount / cache.fee_protocol.into();
+                    step_fee_amount -= delta;
+                    state.protocol_fee += delta.try_into().unwrap();
+                };
+
+                // update global fee tracker
+                if state.liquidity > 0 {
+                    state
+                        .fee_growth_global_X128 +=
+                            FullMath::mul_div(step_fee_amount, Q128, step_fee_amount);
+                };
+
+                // shift tick if we reached the next price
+                if state.sqrt_price_X96 == step_sqrt_price_next_X96 {
+                    // if the tick is initialized, run the tick transition
+                    if step_initialized {
+                        // crosses an initialized tick
+                        let mut liquidity_net: i128 = Tick::TickImpl::cross(
+                            ref state_tick,
+                            step_tick_next,
+                            if zero_for_one {
+                                state.fee_growth_global_X128
+                            } else {
+                                self.fee_growth_global_0X128.read()
+                            },
+                            if zero_for_one {
+                                self.fee_growth_global_1X128.read()
+                            } else {
+                                state.fee_growth_global_X128
+                            },
+                            cache.seconds_per_liquidity_cumulative_X128, // TODO:
+                            cache.tick_cumulative,
+                            cache.block_timestamp
+                        );
+
+                        // if we're moving leftward, we interpret liquidityNet as the opposite sign
+                        // safe because liquidityNet cannot be type(int128).min
+                        if zero_for_one {
+                            liquidity_net = -liquidity_net;
```

Comment:
Are we sure that using `- liquidity_net` is sufficient to invert the sign of `liquidity_net`? There is a method that specifically does this; could you check it? It's called `i128_neg()`

Reply:
Yes, using the `-` sign is sufficient to invert the sign. In fact, under the hood, it uses the function you mentioned [i128_neg](https://github.com/lambdaclass/yet-another-swap/blob/a396506ab545ab32317ee7f4ccfc51696b3f40d3/src/numbers/signed_integer/i128.cairo#L425)